treenode_test.go

```diff
@@ -41,6 +43,58 @@ func TestHandlerReturn(t *testing.T) {
 	assert.NotNil(t, p.RegisterHandler(p.HandlerError3))
 }
 
+type dummyMsg struct{}
+
+type configProcessor struct {
+	configCount int
+	expected    int
+	done        chan<- bool
+}
+
+func (p *configProcessor) Process(env *network.Envelope) {
+	if env.MsgType == ConfigMsgID {
+		p.configCount++
+		if p.configCount == p.expected {
+			p.done <- true
+		}
+	}
+}
+
+func TestConfigPropagation(t *testing.T) {
+	local := NewLocalTest()
+	defer local.CloseAll()
+	const treeSize = 3
+	var serviceConfig = []byte{0x01, 0x02, 0x03, 0x04}
+	hosts, _, tree := local.GenTree(treeSize, true)
+	_, err := hosts[0].overlay.CreateProtocol(spawnName, tree, NilServiceID)
+	log.ErrFatal(err)
+
+	done := make(chan bool)
+	pr := &configProcessor{expected: treeSize - 1, done: done}
+
+	for _, host := range hosts {
+		host.RegisterProcessor(pr,
+			ProtocolMsgID,
+			RequestTreeMsgID,
+			SendTreeMsgID,
+			RequestRosterMsgID,
+			SendRosterMsgID,
+			ConfigMsgID)
+	}
+
+	network.RegisterMessage(dummyMsg{})
+	rootInstance, _ := local.NewTreeNodeInstance(tree.Root, spawnName)
+	rootInstance.SetConfig(&GenericConfig{serviceConfig})
```

Comment:
Needs to verify `err == nil`.
Add a second line with `SetConfig` and verify `err != nil`.